aggregator/src/datastore.rs

```diff
@@ -879,20 +879,14 @@ impl<C: Clock> Transaction<'_, C> {
         )?)
     }
 
-    /// get_client_report retrieves a client report by ID.
+    /// Retrieve a client report by ID, but without decoding the row's columns into VDAF-specific
+    /// types
     #[tracing::instrument(skip(self), err)]
-    pub async fn get_client_report<const L: usize, A>(
+    async fn get_encoded_client_report(
```

Comment:
Why do we need `EncodedClientReport`? If it's just to check equality, we can define equality on `LeaderStoredReport` & continue using that. If it's to avoid re-parsing on the slow read path, I'm not sure this is worth the code complexity vs just parsing the existing report into a `LeaderStoredReport`.

In any case, the pattern used throughout `datastore.rs` is for places that need to "share" parsing logic to use `foo_from_row` helper methods that take a `Row` (and other values, if already parsed everywhere they are needed). Let's follow that here.